File: src/data/coachingQuestions.ts
import { CoachingQuestion } from '../types';

export const coachingQuestions: CoachingQuestion[] = [
  // Realistic Type Questions
  {
    id: 'r1',
    category: 'exploration',
    question: 'What hands-on activities or practical tasks energize you most in your current role?',
    riasecFocus: 'realistic',
    purpose: 'Identify practical strengths and preferences for realistic types',
    followUp: [
      'How could you incorporate more of these activities into your work?',
      'What tools or equipment do you most enjoy working with?'
    ]
  },
  {
    id: 'r2',
    category: 'development',
    question: 'What technical skills would you like to develop to become more effective in practical problem-solving?',
    riasecFocus: 'realistic',
    purpose: 'Focus on skill development for hands-on learners',
    followUp: [
      'What resources or training would help you develop these skills?',
      'How would these skills impact your daily work?'
    ]
  },
  {
    id: 'r3',
    category: 'goal-setting',
    question: 'What concrete, measurable outcomes would you like to achieve in the next 6 months?',
    riasecFocus: 'realistic',
    purpose: 'Help realistic types set tangible, achievable goals',
    followUp: [
      'What specific steps will you take to reach these outcomes?',
      'How will you measure your progress?'
    ]
  },

  // Investigative Type Questions
  {
    id: 'i1',
    category: 'exploration',
    question: 'What complex problems or research challenges excite you most?',
    riasecFocus: 'investigative',
    purpose: 'Identify analytical interests and intellectual curiosity',
    followUp: [
      'What methods do you use to approach these challenges?',
      'How do you stay current with developments in your field?'
    ]
  },
  {
    id: 'i2',
    category: 'development',
    question: 'What areas of knowledge or expertise would you like to deepen through independent study?',
    riasecFocus: 'investigative',
    purpose: 'Support continuous learning and specialization',
    followUp: [
      'What learning resources would be most valuable to you?',
      'How would this deeper knowledge benefit your work?'
    ]
  },
  {
    id: 'i3',
    category: 'reflection',
    question: 'How do you prefer to analyze information and make decisions?',
    riasecFocus: 'investigative',
    purpose: 'Understand analytical thinking processes',
    followUp: [
      'What data or evidence do you find most compelling?',
      'How do you handle uncertainty in your analysis?'
    ]
  },

  // Artistic Type Questions
  {
    id: 'a1',
    category: 'exploration',
    question: 'When do you feel most creative and innovative in your work?',
    riasecFocus: 'artistic',
    purpose: 'Identify creative strengths and optimal conditions',
    followUp: [
      'What environments or situations spark your creativity?',
      'How do you capture and develop your creative ideas?'
    ]
  },
  {
    id: 'a2',
    category: 'development',
    question: 'What creative skills or artistic abilities would you like to develop further?',
    riasecFocus: 'artistic',
    purpose: 'Support creative growth and expression',
    followUp: [
      'How could these skills enhance your professional work?',
      'What creative projects would you like to pursue?'
    ]
  },
  {
    id: 'a3',
    category: 'goal-setting',
    question: 'How can you bring more creativity and innovation to your current role?',
    riasecFocus: 'artistic',
    purpose: 'Integrate creativity into professional goals',
    followUp: [
      'What barriers might prevent creative expression?',
      'How can you advocate for more creative opportunities?'
    ]
  },

  // Social Type Questions
  {
    id: 's1',
    category: 'exploration',
    question: 'What aspects of helping or working with others bring you the most satisfaction?',
    riasecFocus: 'social',
    purpose: 'Identify interpersonal strengths and motivations',
    followUp: [
      'How do you prefer to support and help others?',
      'What impact do you want to have on people?'
    ]
  },
  {
    id: 's2',
    category: 'development',
    question: 'What interpersonal or communication skills would you like to strengthen?',
    riasecFocus: 'social',
    purpose: 'Develop people-focused capabilities',
    followUp: [
      'How would these skills improve your relationships?',
      'What opportunities do you have to practice these skills?'
    ]
  },
  {
    id: 's3',
    category: 'reflection',
    question: 'How do you build trust and rapport with different types of people?',
    riasecFocus: 'social',
    purpose: 'Explore relationship-building strategies',
    followUp: [
      'What challenges do you face in building relationships?',
      'How do you adapt your communication style?'
    ]
  },

  // Enterprising Type Questions
  {
    id: 'e1',
    category: 'exploration',
    question: 'What leadership opportunities or challenges energize you most?',
    riasecFocus: 'enterprising',
    purpose: 'Identify leadership interests and strengths',
    followUp: [
      'What leadership style feels most natural to you?',
      'How do you motivate and influence others?'
    ]
  },
  {
    id: 'e2',
    category: 'goal-setting',
    question: 'What ambitious goals would you like to pursue in your career?',
    riasecFocus: 'enterprising',
    purpose: 'Support goal-oriented and achievement-focused planning',
    followUp: [
      'What resources or support do you need to achieve these goals?',
      'How will you measure success?'
    ]
  },
  {
    id: 'e3',
    category: 'development',
    question: 'What business or entrepreneurial skills would enhance your effectiveness?',
    riasecFocus: 'enterprising',
    purpose: 'Develop business acumen and leadership capabilities',
    followUp: [
      'How could you gain experience in these areas?',
      'What mentors or role models inspire you?'
    ]
  },

  // Conventional Type Questions
  {
    id: 'c1',
    category: 'exploration',
    question: 'What organized, systematic work gives you the greatest sense of accomplishment?',
    riasecFocus: 'conventional',
    purpose: 'Identify preferences for structure and organization',
    followUp: [
      'What systems or processes do you find most effective?',
      'How do you maintain quality and accuracy in your work?'
    ]
  },
  {
    id: 'c2',
    category: 'development',
    question: 'What organizational or administrative skills would you like to improve?',
    riasecFocus: 'conventional',
    purpose: 'Support systematic skill development',
    followUp: [
      'What training or resources would help you develop these skills?',
      'How would these improvements benefit your work?'
    ]
  },
  {
    id: 'c3',
    category: 'reflection',
    question: 'How do you prefer to plan and organize your work for maximum efficiency?',
    riasecFocus: 'conventional',
    purpose: 'Explore organizational preferences and methods',
    followUp: [
      'What tools or systems support your organization?',
      'How do you handle competing priorities?'
    ]
  }
];

File: data/coaching_questions.py
# Coaching Questions Database

coaching_questions = [
    # Realistic Type Questions
    {
        'id': 'r-explore-1',
        'category': 'exploration',
        'riasecFocus': 'realistic',
        'question': 'What hands-on activities or projects have given you the most satisfaction?',
        'purpose': 'Identify specific practical interests and skills',
        'followUp': [
            'What made these experiences particularly rewarding?',
            'How might you incorporate more of these activities into your career?'
        ]
    },
    {
        'id': 'r-develop-1',
        'category': 'development',
        'riasecFocus': 'realistic',
        'question': 'What technical or practical skills would you like to develop further?',
        'purpose': 'Identify skill development opportunities',
        'followUp': [
            'What resources or training would help you develop these skills?',
            'How would mastering these skills impact your career goals?'
        ]
    },
    
    # Investigative Type Questions
    {
        'id': 'i-explore-1',
        'category': 'exploration',
        'riasecFocus': 'investigative',
        'question': 'What problems or mysteries do you find yourself naturally drawn to solving?',
        'purpose': 'Uncover analytical interests and problem-solving preferences',
        'followUp': [
            'What methods do you typically use to investigate these problems?',
            'How could you apply this curiosity in a professional setting?'
        ]
    },
    {
        'id': 'i-develop-1',
        'category': 'development',
        'riasecFocus': 'investigative',
        'question': 'What areas of knowledge would you like to explore more deeply?',
        'purpose': 'Identify learning goals and intellectual interests',
        'followUp': [
            'What specific topics within this area fascinate you most?',
            'How might expertise in this area advance your career?'
        ]
    },
    
    # Artistic Type Questions
    {
        'id': 'a-explore-1',
        'category': 'exploration',
        'riasecFocus': 'artistic',
        'question': 'How do you currently express your creativity, and how would you like to expand this?',
        'purpose': 'Explore creative outlets and artistic aspirations',
        'followUp': [
            'What barriers prevent you from being more creative in your work?',
            'What creative projects would you pursue if resources were unlimited?'
        ]
    },
    {
        'id': 'a-develop-1',
        'category': 'development',
        'riasecFocus': 'artistic',
        'question': 'What creative skills or techniques would enhance your professional value?',
        'purpose': 'Connect creativity to career development',
        'followUp': [
            'How could you practice these skills in your current role?',
            'What creative professionals do you admire and why?'
        ]
    },
    
    # Social Type Questions
    {
        'id': 's-explore-1',
        'category': 'exploration',
        'riasecFocus': 'social',
        'question': 'Describe a time when you made a meaningful difference in someone\'s life.',
        'purpose': 'Identify helping motivations and impact preferences',
        'followUp': [
            'What aspects of helping others energize you most?',
            'How could you create more opportunities for meaningful impact?'
        ]
    },
    {
        'id': 's-develop-1',
        'category': 'development',
        'riasecFocus': 'social',
        'question': 'What interpersonal skills would help you be more effective in supporting others?',
        'purpose': 'Identify relationship and communication skill gaps',
        'followUp': [
            'Which of these skills would have the most immediate impact?',
            'How could you practice these skills in low-stakes situations?'
        ]
    },
    
    # Enterprising Type Questions
    {
        'id': 'e-explore-1',
        'category': 'exploration',
        'riasecFocus': 'enterprising',
        'question': 'What opportunities do you see that others might be missing?',
        'purpose': 'Uncover entrepreneurial thinking and vision',
        'followUp': [
            'What would it take to pursue these opportunities?',
            'What\'s holding you back from taking action?'
        ]
    },
    {
        'id': 'e-develop-1',
        'category': 'development',
        'riasecFocus': 'enterprising',
        'question': 'What leadership or influence skills would accelerate your success?',
        'purpose': 'Identify leadership development needs',
        'followUp': [
            'What leadership challenges do you currently face?',
            'Who are your leadership role models and what can you learn from them?'
        ]
    },
    
    # Conventional Type Questions
    {
        'id': 'c-explore-1',
        'category': 'exploration',
        'riasecFocus': 'conventional',
        'question': 'What systems or processes have you improved or would like to improve?',
        'purpose': 'Identify organizational interests and efficiency focus',
        'followUp': [
            'What inefficiencies frustrate you most in your current work?',
            'How do you approach creating order from chaos?'
        ]
    },
    {
        'id': 'c-develop-1',
        'category': 'development',
        'riasecFocus': 'conventional',
        'question': 'What organizational or analytical tools would make you more effective?',
        'purpose': 'Identify system and process skill needs',
        'followUp': [
            'Which tools or methods have you been curious about?',
            'How would mastering these tools impact your daily work?'
        ]
    }
]

def get_coaching_questions(riasec_type, category=None):
    """Get coaching questions filtered by RIASEC type and optionally by category"""
    questions = [q for q in coaching_questions if q['riasecFocus'] == riasec_type]
    
    if category:
        questions = [q for q in questions if q['category'] == category]
    
    return questions
